src/enclave/enclave_oe.c

```diff
@@ -234,9 +244,11 @@ void sgxlkl_ethread_init(void)
     }
 
     /* Initialization completed, now run the scheduler */
+    a_inc(&ethreads_minus_1);
```

Comment:
I think there’s a race here. What happens if we call this just after another ethread has exited and deleted the enclave state? This won’t happen in normal operation, but an attacker can trigger it.

I think we want to add a second atomic counter on the number of exited threads and increment it before decrementing this counter on thread exit. At this point, we want to check if the exited threads counter is >0 and, if so, exit.